hadoop-ozone/pom.xml

```diff
@@ -147,6 +148,12 @@
         <artifactId>hdds-server-framework</artifactId>
         <version>${hdds.version}</version>
       </dependency>
+      <dependency>
+        <groupId>org.apache.ozone</groupId>
+        <artifactId>hdds-server-framework</artifactId>
+        <version>${hdds.version}</version>
+        <type>test-jar</type>
+      </dependency>
```

Comment:
Not really familiar with this, but is this "test-jar" for hdds-server-framework necessary?

Reply:
Thanks @ivandika3 for the review.

> is this "test-jar" for hdds-server-framework necessary?

It is required for the `MockSpaceUsageCheckFactory$None` configured in `ozone-site.xml`.  Removed now, since it's not essential for the S3 tests, but had to add the normal jar as dependency.  Also added some other used undeclared ones.  And filed HDDS-12689 to add some missing ones for SDK v2.

> After this is merged, HDDS-11879 will be fixed?

We will need some additional changes to fix HDDS-11879.

Reply:
Anyway, as the S3 integration tests are running as intended, so this patch is good to merge.

Reply:
> but the `test-jar` dependency is still there

`hdds-server-framework` no longer has `test-jar`, `hdds-test-utils` does.

https://github.com/apache/ozone/blob/c1a48b57a33753d3e2dc627d2efd1aa0f3eff7ee/hadoop-ozone/integration-test-s3/pom.xml#L59-L69

Reply:
Ok thanks for the clarification, didn't see the changes properly.

I was originally wondering the addition the `hdds-server-framework` `test-jar` dependency on the parent `hadoop-ozone` parent module. Is this because the all the children modules require this `hdds-server-framework` `test-jar` dependency?

Reply:
Oh, sorry, I missed that you referenced the parent POM.  So that one declares it in the `dependencyManagement`, just making it easier to use in child modules.  It does not add the actual dependency in all children.

Reply:
Yes.  Now that the new module does not use it, this is not strictly needed, but I think we can keep it, and later remove `version` from `integration-test` for the same artifact.